app/middleware/headercert/middleware_test.go

```diff
@@ -0,0 +1,253 @@
+package headercert
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/aileron-gateway/aileron-gateway/kernel/testutil"
+	utilhttp "github.com/aileron-gateway/aileron-gateway/util/http"
+)
+
+const (
+	certPath           = "../../../_example/header-cert/pki/client.crt"
+	fpPath             = "../../../_example/header-cert/pki/fingerprint.txt"
+	failCertPath       = "../../../_example/header-cert/pki/fail-client.crt"
+	failFpPath         = "../../../_example/header-cert/pki/fail-fingerprint.txt"
+	expiredCertPath    = "../../../_example/header-cert/pki/expired-client.crt"
+	expiredFpPath      = "../../../_example/header-cert/pki/expired-fingerprint.txt"
+	incompleteCertPath = "../../../_example/header-cert/pki/incomplete-client.crt"
+	rootCAPath         = "../../../_example/header-cert/pki/rootCA.crt"
+)
+
+func TestMiddleware(t *testing.T) {
+
+	type condition struct {
+		method  string
+		headers map[string]string
+	}
+
+	type action struct {
+		status int
+	}
+
+	tb := testutil.NewTableBuilder[*condition, *action]()
+	tb.Name(t.Name())
+	table := tb.Build()
+
+	cert, err := os.ReadFile(certPath)
```

Comment:
No need to check err for test data.
Or, use 'Must' function.

```suggestion
	cert, _:= os.ReadFile(certPath)
```